### ISSUE TITLE
chart_revision_suggester tool sometimes overwrites timeline times incorrectly

### ISSUE THREAD

**Issue by danyx23** [owid/etl#719]
This was reported by PabloA in [slack](https://owid.slack.com/archives/C03JTHBPCN5/p1671821507107289). 

![image (3)](https://user-images.githubusercontent.com/2772449/209954509-7a2ed506-29d8-48f8-8a18-eebd1f748862.png)

The issue here is that if the chart_revision_suggester tool updates one variable in a chart that has several then it checks if it should update the timeline range as well. For this it only takes the newly updated variables into account and then modifies the minTime and maxTime values. What it should do instead is take the full range of all variables after translation into account so that if only one out of 5 variables was updated, the min and max times are only changed if for all of them the the old time slider values are now outside the variables time range.

The offending code seems to be these https://github.com/owid/etl/blob/ecc1fbad39676b220a990f441da2c543dc7c0331/etl/chart_revision_suggester.py#L469-L490

**Comment by danyx23** [comment]
I just realized that this other, related issue has the same cause. In the screenshot below you see a scatter plot turning into a time scatter but that is also because what used to be a single time point selection got turned into a range. The offending code for this one is probably a bit higher up than the code fragment above but in the same function.
![image (4)](https://user-images.githubusercontent.com/2772449/209955551-1a95960d-e55f-488b-85f9-3e84ddc7f801.png)

**Comment by lucasrodes** [comment]
I detected a bug due to `maxtime` in Grapher config not being explicit (default value is `"latest"`). This prevented the chart revision from detecting that the time slider was anchored at a specific year (and not a year range).

This should have been fixed in https://github.com/owid/etl/pull/776.

However, [I have encouraged team members](https://owid.slack.com/archives/C025BDNS5AA/p1673963127233039) to try it out by using the tool in "Experimental mode" (will use the new backend code).
